### PR TITLE
[GSoC] Conversation for required @Argument

### PR DESCRIPTION
**Pull request by KeithBird**
Solution to the issue #453.

This PR is used to track progress and view differences to make discussions easier.

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](https://github.com/apple/swift-argument-parser/blob/main/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by KeithBird** [comment]
We should merge the #451 first.

**Comment by rauhul** [comment]
is interactive parsing unconditional or an opt-in feature via CommandConfiguration?

**Comment by KeithBird** [comment]
I think interactive mode should be turned on by default and let the user have it with no extra work. But, it can be manually turned off via CommandConfiguration. Let's wait and see how the interactive mode ends up, and then decide it when [apple:feature/interactive](https://github.com/apple/swift-argument-parser/tree/feature/interactive) is merged into main.

**Comment by KeithBird** [comment]
Please review @natecook1000

File: Sources/ArgumentParser/Parsing/Interactive.swift [hunk 1]
```diff
@@ -55,22 +55,26 @@ extension CommandParser {
...
+      let name = definition.names.first // (where: { $0.case == .long } )
+      
+      let input: [String]
+      let allValues = definition.help.allValues
+      if allValues.isEmpty {
+        // Get normal value
+        input = getNormalValue(label: label)
+      } else {
+        // Get CaseIterable Enum
+        allValues.enumerated().forEach { print("\($0 + 1). \($1)") }
```
[the remark this answers is not in the document]
Comment (KeithBird) [reply Sources/ArgumentParser/Parsing/Interactive.swift:71]: as UI:
```
1. foo 
2. bar 
3. baz
? Please select an 'mode':
```

File: Sources/ArgumentParser/Parsing/Interactive.swift [hunk 2]
```diff
@@ -80,4 +84,53 @@ extension CommandParser {
     
     return false
   }
+  
+  fileprivate mutating func getNormalValue(label: String) -> [String] {
+    print("? Please enter '\(label)': ", terminator: "")
+    
+    if let input = lineStack?.removeLast().components(separatedBy: " ") {
```
[the remark this answers is not in the document]
Comment (KeithBird) [reply Sources/ArgumentParser/Parsing/Interactive.swift]: Good idea

**Pull request closed without merging**